### ISSUE TITLE
LSPEndpointTest fails

### ISSUE THREAD

**Issue by mschaefer88** [eclipse/lsp4j#398]
I just cloned the repo and ran the build through gradle but the test _LSPEndpointTest_ seems to fail.
Is that an error on my end?

```
org.junit.ComparisonFailure: expected:<...$.result.contents[0][
]
Lists must not cont...> but was:<...$.result.contents[0][]
Lists must not cont...>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.eclipse.lsp4j.test.LSPEndpointTest.testIssue152(LSPEndpointTest.java:182)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:118)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:175)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:157)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:748)```

**Comment by jonahgraham** [comment]
It looks like it is on your end. Perhaps a different formatting of error messages in the comparison due to different versions of dependencies? It passes on build machine https://ci.eclipse.org/lsp4j/job/lsp4j-multi-build/job/master/10/testReport/org.eclipse.lsp4j.test/LSPEndpointTest/ (I just reran the build to make sure)

**Comment by jonahgraham** [comment]
Are you on Windows? The difference looks like line ending differences. The test should be fixed to be stable against such failure.

**Comment by mschaefer88** [comment]
Yes I'm on Windows. Thanks for your input I'll take a look!

**Comment by mschaefer88** [comment]
I fixed both tests that had issues in commit 8ab1388fa9f7df7824ef3f8c3d39636ae59d01bb.

**Comment by jonahgraham** [comment]
> I fixed both tests that had issues in commit [8ab1388](https://github.com/eclipse/lsp4j/commit/8ab1388fa9f7df7824ef3f8c3d39636ae59d01bb).

PR coming? It looks like a trivial review and merge once it does.

BTW the link that github auto-created confused me as it is `https://github.com/eclipse/lsp4j/commit/8ab1388fa9f7df7824ef3f8c3d39636ae59d01bb` - i.e. a commit in the eclipse/lsp4j repo, even though it isn't in this repo.

**Comment by mschaefer88** [comment]
I'll separate it from my current branch and put it into its own and file the pr.

> BTW the link that github auto-created confused me as it is https://github.com/eclipse/lsp4j/commit/8ab1388fa9f7df7824ef3f8c3d39636ae59d01bb - i.e. a commit in the eclipse/lsp4j repo, even though it isn't in this repo.
Yeah, pretty strange, don't know how github normally handles this though. It's a commit in my fork.

**Comment by mschaefer88** [comment]
Opened a pr: mschaefer88:bugfix/tests-failing-on-windows

**Comment by jonahgraham** [comment]
Thank you for the fix in PR #399
